### ISSUE TITLE
Merge tables with self generated auto increment column doesn't show-up Histogram.

### ISSUE THREAD

**Issue by ghost** [buckaroo-data/buckaroo#207]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/paddymul/buckaroo) of Buckaroo.

### What type of jupyter notebook were you using (VSCode notebook, google colab, Jupyter Lab, Jupyter notebook).  Select multiple if you can reproduce this in multiple environments.  If other, please add to description.

Jupyter Notebook

### Reproducible example

```python
df = pd.concat([df1, df2.set_index(df1.index)], axis=1)
```

### Issue description

<img width="774" alt="1" src="https://github.com/paddymul/buckaroo/assets/79505893/305cbb7b-b200-4cd4-b285-b3afecbe53d1">

### Expected behavior

Should show up the histogram for Merge tables with self generated auto increment columns.

### Installed versions

<details>
```
Selected Jupyter core packages...
executing in jupyter-notebook 
buckaroo         : 0.5.0
jupyterlab       : 3.6.3
notebook         : 7.0.6
ipywidgets       : 8.1.1
traitlets        : 5.7.1
jupyter_core     : 5.3.0
pandas           : 2.0.3
numpy            : 1.26.0
IPython          : 8.12.0
ipykernel        : 6.25.0
jupyter_client   : 7.4.9
jupyter_server   : 2.8.0
nbclient         : 0.5.13
nbconvert        : 6.5.4
nbformat         : 5.9.2
qtconsole        : 5.4.2
buckaroo         : C:\Users\91949\anaconda\lib\site-packages\buckaroo\__init__.py
jupyterlab       : C:\Users\91949\anaconda\lib\site-packages\jupyterlab\__init__.py
notebook         : C:\Users\91949\anaconda\lib\site-packages\notebook\__init__.py
ipywidgets       : C:\Users\91949\anaconda\lib\site-packages\ipywidgets\__init__.py
traitlets        : C:\Users\91949\anaconda\lib\site-packages\traitlets\__init__.py
jupyter_core     : C:\Users\91949\anaconda\lib\site-packages\jupyter_core\__init__.py
pandas           : C:\Users\91949\anaconda\lib\site-packages\pandas\__init__.py
numpy            : C:\Users\91949\anaconda\lib\site-packages\numpy\__init__.py
IPython          : C:\Users\91949\anaconda\lib\site-packages\IPython\__init__.py
ipykernel        : C:\Users\91949\anaconda\lib\site-packages\ipykernel\__init__.py
jupyter_client   : C:\Users\91949\anaconda\lib\site-packages\jupyter_client\__init__.py
jupyter_server   : C:\Users\91949\anaconda\lib\site-packages\jupyter_server\__init__.py
nbclient         : C:\Users\91949\anaconda\lib\site-packages\nbclient\__init__.py
nbconvert        : C:\Users\91949\anaconda\lib\site-packages\nbconvert\__init__.py
nbformat         : C:\Users\91949\anaconda\lib\site-packages\nbformat\__init__.py
qtconsole        : C:\Users\91949\anaconda\lib\site-packages\qtconsole\__init__.py
```
</details>

**Comment by paddymul** [comment]
Can you please supply the original dataframes that are being merged?  Or a minimally reproducable testcase.

I can't reproduce this because `df1` and `df2` aren't included.

**Comment by ghost** [comment]
This should be enough i guess.
<img width="446" alt="27" src="https://github.com/paddymul/buckaroo/assets/79505893/1e78bb32-045a-46b0-9f78-7db781c7da7c">

**Comment by paddymul** [comment]
```
import numpy as np
df1 = pd.DataFrame(np.random.randn(8,3))
df2 = pd.DataFrame(np.random.randn(8,3))
df = pd.concat([df1, df2.set_index(df1.index)], axis=1)
df
```

**Comment by paddymul** [comment]
this is caused by non-unique column names.

It could be tested with `df.columns.is_unique`.  I'm not sure where that would fit into the buckaroo processing pipeline.

The correct behavior should be to throw an error and say "Buckaroo needs a dataframe with a unique set of column names"

there is no main key produced in the df_data_dict
```python
bw = buckaroo.BuckarooWidget(df)
bw.df_data_dict
```

I think these types of DataFrames are fairly rare, and since testing for this is quite complex and out of the norm of the processing flow, that I'm going to punt for a while
<img width="555" alt="Screenshot 2024-11-01 at 12 49 14 PM" src="https://github.com/user-attachments/assets/0173db74-eeac-457e-8e9d-3ffabc4ec346">

**Comment by paddymul** [comment]
Fixed by #325   this was caused by a non unique column name list.  Also could have been due to numeric column names.  Numeric column names are coerced to strings.

a dataframe with non unique column names now throws an exception of `DuplicateColumnsException`
